Prototype: Post Settings

kean opened the pull request:
<img src="https://github.com/user-attachments/assets/737cb2e2-404f-4e96-bff9-741c66a6aada" width="320px">

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 500 lin...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kean committed:
Feature: Jetpack Activity Logs (#24597)

* Add PaginatedResponse

* Refactor PaginatedResponse to DataViewPaginatedResponse

- Rename PaginatedResponse to DataViewPaginatedResponse for better naming consistency
- Move files from Pagination to DataView directory
- Update documentation to focus on UI usage with PaginatedForEach
- Add comprehensive unit tests for DataViewPaginatedResponse

* Fix deleteItem to update total count

- Update deleteItem method to properly decrement total when removing items
- Update tests to verify total count is correctly maintained after deletion

* Rename LoadMoreFooterView

* Add DataViewPaginatedForEach component

* Add DataViewPaginatedResponseProtocol

* Refactor subscribers to use DataViewPaginatedResponse

Replace custom SubscribersPaginatedResponse with generic DataViewPaginatedResponse system. Add notification-based subscriber deletion updates and improve pagination handling with DataViewPaginatedForEach component.

* Refactor subscribers service creation and move deleteSubscriber extension

- Rename getSubscribersService() to maketSubscribersService() to better reflect factory pattern
- Move deleteSubscriber extension from SubsriberDetailsViewModel to dedicated SubscribersServiceRemote+Extensions file
- Update all callers to use the renamed method

* Fix typo in Subscriber

* Fix formatting

* Fix typos and address comments

* Implement new Jetpack Activity Logs screen

* Fix SwiftLint warnings

* Add Miniature app

* Add XcodeTarget_App as a dependency

* Configure xcconfig (just use Reader)

* Add initial ActivityLogDetailsView

* Add ActivityLogDetailsView to Miniature target

- Implement modern SwiftUI version of Activity Detail screen
- Use card-based design system matching SubscriberDetailsView
- Add proper localization with NSLocalizedString and reverse-DNS keys
- Use WPStyleGuide for consistent icon and color handling
- Include activity stats visualization for backup events
- Add restore and download backup actions with confirmation dialogs
- Integrate with existing Activity model and RewindStatus
- Update ContentView with navigation links for testing
- Configure MiniatureApp with proper navigation and theming

* Extract Activity icon methods from WPStyleGuide to Activity extension

- Create Activity+Icon.swift with gridiconType, icon, and statusColor properties
- Update all references to use new Activity extension methods
- Deprecate old WPStyleGuide methods with proper deprecation messages
- Remove stringToGridiconTypeMapping from WPStyleGuide
- Fix linter warnings for trailing whitespace and shorthand optional binding

* Remove rewindStatus parameter from ActivityLogDetailsView

- Remove rewindStatus parameter from ActivityLogDetailsView initializer
- Remove rewindStatus checks in ActivityActionsCard
- Update preview providers to not pass rewindStatus
- Update ContentView to not create mock RewindStatus instances
- Remove warning card for multisite installations
- Simplify restore button enable/disable logic

* Dev

* Add formattedContent

* Move the code to the main target

* Extract reusable ActivityActorAvatarView

* Remove unused code and integrate ActivityLogDetailsView

* Fix SwiftLint warnings

* Add initial restore/download backup code

* Remove JetpackSiteRef usages

* Add multisite handling

* Remove the new screens and use the existing flows

* Add analytics

* Cleanup

* Add missing analytics

* Cleanup

* Fix an issue with not all restorable acitivies shown in backups

* Fix an issue with restore/download flow layout

* Show restore checkpoint in section

* Extract reusable CardView and InfoRow components from SubscribersDetailsVIew

* Cleanup

* Add rewindable

* Move restore buttons higher

* Add date filtering back

* Add Backup tracking

* Fix clear background in restore flows

* Rework how we do polling

* Rework how we manage backup statuss

* Revert hasBackup change

* Update tests

* Update release notes

* Remove unused code

* Cleanup

* Remove Application from list

* Remove BackupListViewController and use ActivityLogsViewController instead

- Delete BackupListViewController.swift and its extension
- Update showBackup() to use ActivityLogsViewController with isBackupMode
- Remove BackupListViewController reference from ActivityDetailViewController
- Implement displayBackupWithSiteID using ActivityLogsViewController
- Add WPAnalytics tracking to ContentCoordinator backup display

* Remove JetpackActivityLogViewController and use ActivityLogsViewController

- Delete JetpackActivityLogViewController.swift
- Update showActivity() methods to use ActivityLogsViewController directly
- Update ActivityDetailViewController to check for ActivityLogsViewController
- Simplify DashboardActivityLogCardCell to use ActivityLogsViewController

* Remove dataViews feature flag

- Remove dataViews case from FeatureFlag enum
- Remove dataViews from enabled property switch statement
- Remove dataViews description

The feature flag was already removed from usage in previous commits
where we removed JetpackActivityLogViewController and BackupListViewController.

* Create separate BackupsViewController and remove isBackupMode from ActivityLogsViewController

- Create new BackupsViewController and BackupsView that reuses ActivityLogsView
- Remove isBackupMode parameter from ActivityLogsViewController
- Update all usages to use BackupsViewController for backups
- Update ActivityDetailViewController to recognize BackupsViewController presenter
- Keep ActivityLogsView and ActivityLogsViewModel unchanged as implementation details

* Remove ActivityListViewModelTests

* Remove ActivityTypeSelectorViewController

* Remove ActivityListViewModel

* Remove BaseActivityListViewController

* Fix compilation

* Remove RewindStatusRow

* Remove RewindStatusTableViewCell

* Remove ActivityListRow

* Remove ActivityTableViewCell

* Remove CalendarViewController

* Remove JTAppleCalendar dependency

* Remove FilterBarView

* Remove FilterChipButton

* Remove ActivityDetailViewController

* Remove ActivityStore usages

* Remove ActivityStoreTests

* Fix how isAwaitingCredentials works

* Remove ActivityStore

* Integrate ActivityContentRouter and FormattableActivity

* Update filter icon

* Update filters icon

* Remove WPStyleGuide+Activity and fix an issue with ActivityFormattableContentView layout

* Fix layout of ActivityFormattableContentView

* Add placeholders for empty fields

* Remove build instructions from CLAUDE.md

* Add more instructions for CLAUDE

* Handle a scenario where logs are from existing user

* Remove obsolete tests

* Update UI tests

* Fix release build

* Use Duration

* Simplify CardView

* Add TODO for iOS 17

* Move state change to onAppear
